lib/add_magic_comment.rb

```diff
@@ -16,11 +16,20 @@ module AddMagicComment
 
   def self.process(argv)
     directory = argv.first || Dir.pwd
+    ignore_pattern = Regexp.new(argv[1]) rescue nil
+    # Allow providing as either a Regex or a String
+    if ignore_pattern && ignore_pattern.is_a?(String)
```

Comment:
Yeah, I think passing an actual Regex is only feasible if called programmatically (e.g. I was testing via IRB). The one nice thing having the ability to pass directly gets you though is being able to provide a fully configured Regex object to do things like case-insensitive matching, etc.